v8isolate.cc

```diff
@@ -14,14 +14,31 @@ void* ArrayBufferAllocator::AllocateUninitialized(size_t length) {
 
 void ArrayBufferAllocator::Free(void* data, size_t) { free(data); }
 
-V8Isolate::V8Isolate() {
+V8Isolate::V8Isolate() : startup_data{} {
   v8::Isolate::CreateParams create_params;
   create_params.array_buffer_allocator = &allocator;
   isolate_ = v8::Isolate::New(create_params);
 }
 
+V8Isolate::V8Isolate(const char* snapshot_js_str) {
+  startup_data = v8::V8::CreateSnapshotDataBlob(snapshot_js_str);
+  if (startup_data.data == NULL) {
+    throw BadSnapshotDataException();
```

Comment:
Part of the reason is that the snapshot creation and the isolate construction are combined in this API.  In my fork/rewrite, I explicitly made these separate all the way up to the Go API, but here you could at least make the C++ bridge API have two steps and catch those explicitly on the Go side.  That way you could potentially provide a more useful error...

...Except that snapshot blobs don't fail if the JS is bad.  So..... the exception will basically never get called unless you're out of memory or something, I think.  Anyways, I'm OK with it too, but would slightly prefer two calls in v8_create_isolate_with_snapshot